firedrake/checkpointing.py

```diff
@@ -288,3 +289,161 @@ def __del__(self):
         if hasattr(self, "comm"):
             free_comm(self.comm)
             del self.comm
+
+
+class HDF5File(object):
+
+    """An object to mimic the DOLFIN HDF5File class.
```

Comment:
Sure. That information was probably better incorporated in the pull request text.

In any case, I wanted to justify the choice of kwarg names, as it's not firedrake-standard or even consistent (e.g. filename vs file_mode); unfortunately it's necessary to use the same code with both dolfin and firedrake backends.